lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -741,6 +797,131 @@ class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
 };
 } // namespace
 
+// productDimSize = product(size(dim) for dim in dims)
+// aten.mean(x, dims) = aten.sum(x, dims) / productDimSize.
+namespace {
+class DecomposeAtenMeanDimOp : public OpRewritePattern<AtenMeanDimOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanDimOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value dimList = op.dim();
+    Value keepDim = op.keepdim();
+    Type outputType = op.getType();
+
+    BaseTensorType inputType = input.getType().cast<BaseTensorType>();
+    if (!inputType.hasDtype() || !inputType.getDtype().isa<mlir::FloatType>()) {
+      return rewriter.notifyMatchFailure(
+          op, "only floating-point type is supported");
+    }
+
+    // Compute sum along dimensions specified in `dimList`.
+    Value none = rewriter.create<Torch::ConstantNoneOp>(loc);
+    Value sumAlongDims = rewriter.create<AtenSumDimIntListOp>(
+        loc, outputType, input, dimList, keepDim, /*dtype=*/none);
```

Comment:
`AtenMeanDimOp` has a `dtype` argument that should be passed along into `AtenSumDimIntListOp`